moto/config/aws_managed_rules.py

```diff
@@ -0,0 +1,200 @@
+"""List of IDs for AWS Config Managed rules.
+
+Started with the list found here, then went to each page to get the ID:
```

Comment:
Could we write a script to get this list from boto3? Would be good to have something reproducible, in case this list will change in the future

Reply:
I agree, it would be nice to be able to get the latest list, but I couldn't find a way to do so.  It don't see any APIs for the Config Service that would get that list.  boto3 must have the list for them to perform their validation checks, but if I pick one of those IDs and search through boto3's github repo, it's not finding a match.  There are yaml files related to best practices that list IDs, but they don't have the complete list and refer you to the page where I got the list.

I have a lot more work to do for this PR, but I wanted to commit what I had so far to ensure I didn't accidently lose or delete what I have so far.

Reply:
@bblommers This just got a lot worse.  I was experimenting a bit and the AWS managed rule constrains the value of the InputParameters and MaximumExecutionFrequency arguments.  I would need that info to perform accurate validation.  I really wish I could find an API that would retrieve that info, otherwise I'd need to do some web scraping and create the structure I could use for validation.

Reply:
https://github.com/awsdocs/aws-config-developer-guide/blob/main/doc_source/managed-rules-by-aws-config.md  -- it's the source for the AWS documentation and Managed Rule list.

Reply:
It's a bit awkward, but it is possible to retrieve all rules programmatically - by enabling all rules first in the Console, getting the full list via the API and then deleting them all again.

The output of `describe_config_rules` can be found here - feel free to use this in your PR:
https://gist.github.com/bblommers/0a451593c7473e95ad74c671c952258a

This should contain all AWS managed rules, and their InputParameters/MaximumExecutionFrequency

Steps on how to do it, if you're interested in reproducing the list:
https://gist.github.com/bblommers/91cf995def518f9057a4afe71ca519a4

**Note that you might incur costs, so follow these steps at your own risk**

Reply:
Thank you very much!  I'll use your list.  I had hoped it would have been simple to read the AWS markdown files and then convert them to JSON and work from there, but there's no simple markdown to JSON conversion for Python.  It wouldn't be too difficult to write, but your solution will do just fine.

Reply:
I started working on the list of AWS managed rules and out of curiosity, I compared the IDs in your list with the IDs in the markdown list I had created.  There are 34 more IDs on the managed list than on yours:

```
{'IAM_GROUP_HAS_USERS_CHECK', 'S3_BUCKET_BLACKLISTED_ACTIONS_PROHIBITED', 'CODEBUILD_PROJECT_ENVVAR_AWSCRED_CHECK', 'EC2_MANAGEDINSTANCE_PLATFORM_CHECK', 'IAM_INLINE_POLICY_BLOCKED_KMS_ACTIONS', 'RESTRICTED_INCOMING_TRAFFIC', 'SERVICE_VPC_ENDPOINT_ENABLED', 'S3_BUCKET_POLICY_NOT_MORE_PERMISSIVE', 'ELB_TLS_HTTPS_LISTENERS_ONLY', 'DESIRED_INSTANCE_TYPE', 'IAM_ROLE_MANAGED_POLICY_CHECK', 'ELB_PREDEFINED_SECURITY_POLICY_SSL_CHECK', 'EC2_MANAGEDINSTANCE_APPLICATIONS_REQUIRED', 'CLOUDWATCH_ALARM_ACTION_CHECK', 'API_GW_ENDPOINT_TYPE_CHECK', 'EC2_MANAGEDINSTANCE_APPLICATIONS_BLACKLISTED', 'ACCESS_KEYS_ROTATED', 'REDSHIFT_CLUSTER_CONFIGURATION_CHECK', 'IAM_USER_UNUSED_CREDENTIALS_CHECK', 'REQUIRED_TAGS', 'CLOUDFORMATION_STACK_DRIFT_DETECTION_CHECK', 'APPROVED_AMIS_BY_TAG', 'FMS_WEBACL_RESOURCE_POLICY_CHECK', 'CLOUDWATCH_ALARM_RESOURCE_CHECK', 'EC2_MANAGEDINSTANCE_INVENTORY_BLACKLISTED', 'APPROVED_AMIS_BY_ID', 'LAMBDA_FUNCTION_SETTINGS_CHECK', 'IAM_POLICY_BLACKLISTED_CHECK', 'DESIRED_INSTANCE_TENANCY', 'CODEBUILD_PROJECT_SOURCE_REPO_URL_CHECK', 'FMS_SHIELD_RESOURCE_POLICY_CHECK', 'IAM_POLICY_IN_USE', 'ELB_CUSTOM_SECURITY_POLICY_SSL_CHECK', 'REDSHIFT_CLUSTER_MAINTENANCESETTINGS_CHECK', 'CLOUDWATCH_ALARM_SETTINGS_CHECK', 'IAM_CUSTOMER_POLICY_BLOCKED_KMS_ACTIONS', 'FMS_WEBACL_RULEGROUP_ASSOCIATION_CHECK'}
```

I was looking at the history of the AWS markdown file and it's updated every month or every other month.

I'm going to write a script to pulldown and parse the markdown files.  I'm assuming this script could update or replace the `aws_managed_rules.py` file during a moto install/release?  But then wouldn't it need to be checked in as well?  Could you tell me where I should commit the script and how it should be invoked (or handle that part if you'd like)?  It all seems a bit messy.

Reply:
Hmm.. interesting. Strange, that there are 34 rules not available via the Console. (Either that, or I just missed something)

Yes, it makes sense to have it replace `aws_managed_rules.py` with each invocation.

All of our other scripts are inside the `scripts` folder. None of them are invoked automatically during release at the moment, they're all manual. So I'd be happy to treat this script the same - just check it in inside that folder, and we'll invoke it manually until there's a better way.

>
>  It all seems a bit messy.

Couldn't agree more!

Reply:
Some of these rules are region-specific.  I wasn't planning on checking for that.  

```
**AWS Region:** All supported AWS regions except Asia Pacific \(Osaka\), Europe \(Milan\), Middle East \(Bahrain\) Region
```

Reply:
I've committed the script `pull_down_aws_managed_rules.py` into the scripts directory.  Couple of notes:

- It's slow because it's reading 192 files at the moment.  If run manually, it'd be nice to print the name of the file it's currently reading, but I removed that for now.
- If it detects an error, it prints an error to syserr.  I wasn't sure what else to do.
- It uses a relative path to put the json file into `moto/moto/config`.
- I output the region info, but I currently don't plan to use it.  That's an exercise for someone else!

Feel free to modify this file and make it work as you see fit.  For now I'll use the JSON output to add ConfigRule validations for managed rules.

Reply:
Oops ... there's a bug in the collection of parameters.